### PR TITLE
Fix/dynamic element of collapse in hsection

### PR DESCRIPTION
**Pull request by Thienhuynh95**
## Description

This PR:
- Update `README` for new generating and consuming dynamic constant process
- Improve height detection on expanding behavior of the `Collapse` component
- Add a story for `HSection` about having a delay append item
- Add a hook for element resizing detection

## Todo

- [ ] Bump version and add tag

### CODE REVIEW

File: src/utils/useResizeAware.tsx [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+export type UseResizeAwareProps = [
+  ReactNode,
+  ReturnType<typeof defaultReporter>
+];
+
+/** @resource https://github.com/FezVrasta/react-resize-aware */
+/**
+ * Hook to detect element when it changes in size (height / width)
+ * @NOTE Make sure the measured element to have `position != initial`
```
Comment (goodpickle) [inline src/utils/useResizeAware.tsx:20]:
```suggestion
 * @NOTE Make sure the measured element has `position != initial`
```

File: src/utils/useResizeAware/_ResizeListener.tsx [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  onResize,
+}: {
+  onResize: (ref: MutableRefObject<HTMLIFrameElement | null>) => void;
+}) => {
+  const ref = useRef<HTMLIFrameElement | null>(null);
+
+  const getTarget = () =>
+    ref.current &&
+    ref.current.contentDocument &&
+    ref.current.contentDocument.defaultView;
```
Comment (goodpickle) [inline src/utils/useResizeAware/_ResizeListener.tsx:15]:
```suggestion
  const getTarget = () => ref?.current?.contentDocument?.defaultView;
```

File: build-utils/css/README.md [hunk 3]
```diff
@@ -1,8 +1,112 @@
...
+- Sub script for generating dynamic constants to be consumed in `build-utils/css/constants/theme.ts` process is `build-utils/css/buildLogicConstants.ts`
+- Main script for this whole building process is `build-utils/css/buildTheme.ts`.
+---
+
+## **Generating dynamic constant process overview**
+Sub script `build-utils/css/buildLogicConstants.ts` is primarily created for having a way to generate a constant programmatically with logic and customized jsdocs comments.  
+**REASON**: Typescript will not read and provide jsdocs info on the spread operators (Ex: `...sthing`), so we can't benefit from jsdocs tag when using typescript key suggestion.  
+  
+### **When you run the script, it will**:
+*BEBFORE STARTED, UNTIL WE HAVE A SCRIPT TO GENERATE THE TEMPLATE, YOU MAY NEED TO MANUALLY PUT A FILE WITH SOME REQUIRED FUNCTIONS AND VARIABLES WITH NAMING SYNTAX `_${name}.ts` (WHICH WILL BE EXPLAINED HOW DEEPER BELOW) INTO `build-utils/css/constants/logics`.*
```
Comment (devaneereid) [inline build-utils/css/README.md:13]: This really looks great Thein 🎉   I think there might be a tiny typo on `*BEFORE`?

File: src/utils/useResizeAware/_ResizeListener.tsx [hunk 4]
```diff
@@ -0,0 +1,68 @@
+import React, { MutableRefObject, useCallback, useRef } from 'react';
+
+// This automatically attaches to itself the resize event listener
+// and adds onResize as callback
+export const ResizeListener = ({
+  onResize,
+}: {
+  onResize: (ref: MutableRefObject<HTMLIFrameElement | null>) => void;
+}) => {
+  const ref = useRef<HTMLIFrameElement | null>(null);
```
Comment (goodpickle) [inline src/utils/useResizeAware/_ResizeListener.tsx:10]: What do you think about simplifying the type? `useRef` infers `null`.
```suggestion
  const ref = useRef<HTMLIFrameElement>(null);
```

File: build-utils/css/README.md (hunk 3 above)
Comment (Thienhuynh95) [reply build-utils/css/README.md:13]: Good catch 😆 Thanks for reading and noticing that 😉

File: src/utils/useResizeAware/_ResizeListener.tsx [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+    }
+
+    // clean event listener on unmount
+    return () => {
+      // Ensure the target exists and is in fact an event listener
+      // this fixes an issue where contentDocument.defaultView is not a real window object
+      // as can be the case when used with React portals
+      const target = getTarget();
+      const isListener =
+        target && typeof target.removeEventListener === 'function';
```
Comment (goodpickle) [inline src/utils/useResizeAware/_ResizeListener.tsx:40]:
```suggestion
      const isListener = typeof target?.removeEventListener === 'function';
```

File: src/utils/useResizeAware.tsx [hunk 6]
```diff
@@ -0,0 +1,50 @@
+import React, { ReactNode, useCallback, useMemo, useState } from 'react';
+
+import { ResizeListener } from './useResizeAware/_ResizeListener';
+
+const defaultReporter = (target: HTMLElement | null) => {
+  return {
+    width: target != null ? target.offsetWidth : null,
+    height: target != null ? target.offsetHeight : null,
```
Comment (goodpickle) [inline src/utils/useResizeAware.tsx:8]:
```suggestion
    width: target?.offsetWidth || null,
    height: target?.offsetHeight || null,
```

File: src/utils/useResizeAware/_ResizeListener.tsx [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+    const target = getTarget();
+    if (target) {
+      target.addEventListener('resize', () => onResize(ref));
+    }
+  }, [onResize]);
+
+  React.useEffect(() => {
+    if (getTarget()) {
+      load();
+    } else if (ref.current && ref.current.addEventListener) {
```
Comment (goodpickle) [inline src/utils/useResizeAware/_ResizeListener.tsx:29]:
```suggestion
    } else if (ref.current?.addEventListener) {
```

**Review by goodpickle — CHANGES_REQUESTED** [review]
Looks great! I made some minor code style nitpicks.

**Review by goodpickle — DISMISSED** [review]

**Review by evad1n — APPROVED** [review]

**Pull request merged**